### PR TITLE
Filter topics out based on committer

### PR DESCRIPTION
**Pull request by bharathi-ravishanker**

### CODE REVIEW

**Comment by bharathi-ravishanker** [comment]
Initial commit without tests @denisgrenader

File: src/main/scala/com/box/castle/committer/api/CommitterFactory.scala [hunk 1]
```diff
@@ -112,6 +112,15 @@ trait CommitterFactory extends Closeable {
    */
   def createCommitterExceptionRetryStrategy: RetryStrategy = DefaultRecoverableExceptionRetryStrategy
 
+  /**
+    * Creates a filter which would filter topics that the committer should process
+    *
+    * @return
+    */
+  def createFilter(): Filter = {
```
Comment (denisgrenader) [inline src/main/scala/com/box/castle/committer/api/CommitterFactory.scala]: createTopicFilter(): TopicFilter

File: src/main/scala/com/box/castle/committer/api/FilterTopicsFactory.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.box.castle.committer.api
+
+/**
+  * Created by bravishanker on 7/7/17.
+  */
+class FilterTopicsFactory {
```
Comment (denisgrenader) [inline src/main/scala/com/box/castle/committer/api/FilterTopicsFactory.scala]: I don't think we need a separate factory for this, let's remove this alltogether

File: src/main/scala/com/box/castle/committer/api/Filter.scala [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.box.castle.committer.api
+
+/**
+  * Created by bravishanker on 7/7/17.
+  */
+class Filter {
```
Comment (denisgrenader) [inline src/main/scala/com/box/castle/committer/api/Filter.scala]: class TopicFilter {

File: src/main/scala/com/box/castle/committer/api/Filter.scala [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+  def containsTopic(topic: String): Boolean = {
```
Comment (denisgrenader) [inline src/main/scala/com/box/castle/committer/api/Filter.scala]: def match(topic: String): Boolean

**Review by denisgrenader — CHANGES_REQUESTED** [review]

**Review by denisgrenader — APPROVED** [review]

**Pull request merged**